models/superrare/superrare_ethereum_events.sql

```diff
@@ -0,0 +1,372 @@
+, rare_token_price_eth as ( 
+    with all_rare_eth_dex_trades as (
+        select  block_time
+                , token_bought_symbol
+                , token_sold_symbol
+                , token_bought_amount
+                , token_sold_amount 
+                , case when token_bought_symbol like '%ETH%' then token_bought_amount*1.0/nullif(token_sold_amount,0) 
+                    else  token_sold_amount*1.0/nullif(token_bought_amount,0)
+                end as eth_per_rare
+        -- from dex.trades
+        from {{ ref('dex_trades') }}
+        where   -- RARE trades
+                (token_bought_address = lower('0xba5bde662c17e2adff1075610382b9b691296350') or token_sold_address = lower('0xba5bde662c17e2adff1075610382b9b691296350'))
+            and (token_bought_symbol like '%ETH%' or token_sold_symbol like '%ETH%')    
+        order by block_time desc 
+    )
+    , all_days as (
+        select explode(sequence(to_date((select min(date_trunc('week',block_time)) from all_rare_eth_dex_trades)), to_date(now()), interval 1 week)) as week
+    )
+    
+    select a.week
+            , average_price_that_day_eth_per_rare
+    from all_days a 
+    left outer join (
+        select date_trunc('week',block_time) as week
+                , avg(eth_per_rare) as average_price_that_day_eth_per_rare
+        from (select * from all_rare_eth_dex_trades where eth_per_rare < 0.001) a
+        group by 1 
+        ) b 
+    on a.week = b.week
+)
+
+-- Mint details needed to determine whether a transaction was a primary or secondary sale. 
+-- Sometimes primary sales are not sold directly from the artist's address (e.g. sold from Gallery)
+, mint_address_details_per_token_id as (
+    select  contract_address
+            , `tokenId`
+            , a.`to` as mint_sent_to
+            , b.`from` as mint_created_by
+    from 
+    (
+    select contract_address
+            , `tokenId`
+            , evt_tx_hash
+            , `to` 
+    -- from erc721_ethereum.evt_Transfer
+    from {{ source('erc721_ethereum','evt_transfer') }}
+    where contract_address in (select distinct contract_address from all_superrare_sales)
+    and `tokenId` in (select distinct `_tokenId` from all_superrare_sales)
+    and `from` = '0x0000000000000000000000000000000000000000'
+    ) a 
+    inner join 
+    ( select `from` 
+            , hash 
+    -- from ethereum.transactions
+    from {{ source('ethereum','transactions') }}
+    ) b 
+    on a.evt_tx_hash = b.hash
+)
+, mint_address_per_token_20 as (
+    select contract_address
+            , `value` as `tokenId`
+            , `to` as mint_address 
+    -- from erc20_ethereum.evt_Transfer
+    from {{ source('erc20_ethereum','evt_transfer') }}
+    where contract_address in (select distinct contract_address from all_superrare_sales)
+    and `value` in (select distinct `_tokenId` from all_superrare_sales)
```

Comment:
{% if is_incremental() %} -- this filter will only be applied on an incremental run
    and evt_block_time >= date_trunc("day", now() - interval '1 week')
    {% endif %}